packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -348,6 +346,46 @@ mixin SchedulerBinding on BindingBase, ServicesBinding {
     return entry.completer.future;
   }
 
+  TimingsCallback _oldTimingsCallback;
+  StreamController<FrameTiming> _frameTimingBroadcastController;
+  int _frameTimingListenCount = 0;
+
+  void _onFrameTimingListen() {
+    _frameTimingListenCount += 1;
+  }
+
+  // If there's no one listening, remove `_frameTimingBroadcastController` and
+  // reset [window.onReportTimings] back to `_oldTimingsCallback`. This is to
+  // reduce the overhead if `_oldTimingsCallback` is null. Resetting
+  // [window.onReportTimings] back to null tells the engine that there's no need
+  // to send [FrameTiming] from engine to the framework.
+  void _onFrameTimingCancel() {
+    _frameTimingListenCount -= 1;
+    assert(_frameTimingListenCount >= 0);
+    if (_frameTimingListenCount == 0) {
+      window.onReportTimings = _oldTimingsCallback;
```

Comment:
I'd love to do that. Would that be a breaking change as we're removing `onReportTimings` and only expose this broadcast stream from `dart:ui Window`?

Reply:
It would be, but I think the only use of the API is in package:flutter itself. The breaking change process would be a good way to find any additional usages, and to let people who are using it know how to migrate to the new API (which would be a straightforward migration).